config/routes.rb

```diff
@@ -11,4 +11,5 @@
 
   resources :posts, only: [:show, :new, :create]
 
+  get 'users/:nickname', to: 'users#show', as: 'user', nickname: /[^\/]+/
```

Comment:
The `as` alias creates a named path. Probably we want something like 'show_user', instead of just 'user', so you then could use `show_user_path`